### PR TITLE
UI validation - adding icon mediatype check

### PR DESCRIPTION
**Pull request by galletti94**
check that mediatype is valid

### CODE REVIEW

File: operatorcourier/validate.py [hunk 1]
```diff
@@ -390,6 +390,9 @@ def get_alm_kinds(alm_examples):
             for example in alm_examples:
                 res.append(example["kind"])
             return res
+        
+        def is_mediatype(mediatype):
+            return mediatype in ["image/gif", "image/jpeg", "image/png", "image/svg+xml"]
```
Comment (SamiSousa) [inline operatorcourier/validate.py]: Is this an exhaustive list of our supported image types?

File: operatorcourier/validate.py [hunk 2]
```diff
@@ -469,5 +472,28 @@ def get_alm_kinds(alm_examples):
...
                             "valid capabilities level", annotations["capability"])
             valid = False
+        
+        # icon check
+        if "icon" in spec:
+            icon = spec["icon"]
+            if isinstance(icon, (list,)):
+                if len(icon) == 1:
+                    if len(icon[0].keys()) == 2:
+                        if "base64data" in icon[0] and "mediatype" in icon[0]:
```
Comment (SamiSousa) [inline operatorcourier/validate.py]: Will we also check that the base64data is valid?

File: operatorcourier/validate.py (hunk 1 above)
Comment (galletti94) [reply operatorcourier/validate.py]: as far as I know yes

File: operatorcourier/validate.py (hunk 2 above)
Comment (galletti94) [reply operatorcourier/validate.py]: I don't think that's really possible. I can check that the starting bits correctly match what the mediatype is defined to be but as far as I know there is no way to check that the image is "valid". I suggest just going to operatorhub.io/preview for a visual confirmation

**Pull request merged**